File: JavaScript/convertStringToCamelCase.js
function toCamelCase(str) {
  if (str.length > 0) {
    const words = str.split(/[_-]/);
    let camelCaseString = words[0];

    for (let i = 1; i < words.length; i++) {
      camelCaseString +=
        words[i][0].toUpperCase() + words[i].substr(1).toLowerCase();
    }
    return camelCaseString;
  } else {
    return "";
  }
}

const { assert } = require("chai");

describe("Tests", () => {
  it("test", () => {
    assert.strictEqual(
      toCamelCase(""),
      "",
      "An empty string was provided but not returned"
    );
    assert.strictEqual(
      toCamelCase("the_stealth_warrior"),
      "theStealthWarrior",
      "toCamelCase('the_stealth_warrior') did not return correct value"
    );
    assert.strictEqual(
      toCamelCase("The-Stealth-Warrior"),
      "TheStealthWarrior",
      "toCamelCase('The-Stealth-Warrior') did not return correct value"
    );
    assert.strictEqual(
      toCamelCase("A-B-C"),
      "ABC",
      "toCamelCase('A-B-C') did not return correct value"
    );
  });
});

File: C#/ConvertStringToCamelCase.cs
using System;
using NUnit.Framework;

public class Kata
{
    public static string ToCamelCase(string str)
    {
        string[] words = str.Split('-', '_');
        string camelCase = words[0];

        for (int i = 1; i < words.Length; i++)
        {
            camelCase += char.ToUpper(words[i][0]) + words[i].Substring(1).ToLower();
        }

        return camelCase;
    }
}



[TestFixture]
public class KataTest
{
    [Test]
    public void KataTests()
    {
        Assert.AreEqual("theStealthWarrior", Kata.ToCamelCase("the_stealth_warrior"), "Kata.ToCamelCase('the_stealth_warrior') did not return correct value");
        Assert.AreEqual("TheStealthWarrior", Kata.ToCamelCase("The-Stealth-Warrior"), "Kata.ToCamelCase('The-Stealth-Warrior') did not return correct value");
    }
}